python/src/scipp/table_html/formatting_html.py

```diff
@@ -44,36 +43,33 @@ def _make_row(data_html, variances_html=None):
     return f"<div>{data_html}</div>"
 
 
-def _format_row(data, size):
-    if size == 0:
-        return _make_row("[]")
-    return _make_row(_format_array(data, size))
-
-
 def _format_non_sparse(var, has_variances):
     size = reduce(operator.mul, var.shape, 1)
     # flatten avoids displaying square brackets in the output
     data = retrieve(var, variances=has_variances).flatten()
-    return _format_row(data, size)
+    return _make_row(_format_array(data, size, ellipsis_after=2))
 
 
-def _format_sparse(var, has_variances):
+def _get_sparse(var, variances, ellipsis_after):
     if hasattr(var, "data") and var.data is None:
-        return "no data in sparse array"
-    s = []
+        return ["no data, implicitly 1"]
     size = var.shape[0]
-
     i = 0
     s = []
     while i < size:
-        if i == 2 and size > 5:
-            s.append(_make_row("...", ''))
-            i = size - 2
-        data = retrieve(var, variances=has_variances)[i]
-        s.append(_format_row(data, len(data)))
+        if i == ellipsis_after and size > 2 * ellipsis_after + 1:
+            s.append("...")
+            i = size - ellipsis_after
+        data = retrieve(var, variances=variances)[i]
+        s.append('list(' + _format_array(data, len(data), ellipsis_after) +
```

Comment:
The details should should the full data, not have it elided. The intent is for it to be copy pastable, and now that it's actually wrapped by `array(list(` it would be possible as long as we don't elide the data

Reply:
They are elided for numpy arrays. We *must* elide, otherwise you end up converting 100 GB data to HTML?

Reply:
One one hand that is the point of `__repr__`. Short strings should be given via `__str__`. On the other hand I can see why we don't want to print all of it.

The numpy compromise is print up to length 1000. I think that's sensible and solves both issues - small datasets can be fully `to_html`ed and pretty much reconstructed from that, while big ones can still be visualised

Reply:
Do you have a suggestion on how to determine what to ellide? It is not so trivial because it is not just a 2D array.

Reply:
Keep the ellision in the preview as it is, but in the data repr ellide only arrays over 1000 length maybe?

Reply:
Yes, I just pushed an update. It is a bit ugly because it needs to consider the length of all subarrays.